src/views/feedback/feedback.vue

```diff
@@ -0,0 +1,411 @@
+    <!-- 全部 pagination -->
+    <pagination
+      v-show="isRead === 0"
+      :total="feedBackStore.allFeedBack.total"
+      :page="feedBackStore.allFeedBack.current"
+      :limit="feedBackStore.allFeedBack.size"
+      @pagination="clickToChange"
+      class="pagination_feedBack"
+    />
+    <!-- 已读 pagination -->
+    <pagination
+      v-show="isRead === 1"
+      :total="feedBackStore.readFeedBacks.total"
+      :page="feedBackStore.readFeedBacks.current"
+      :limit="feedBackStore.readFeedBacks.size"
+      @pagination="clickToRead"
+      class="pagination_feedBack"
+    />
+    <!-- 未读 pagination -->
+    <pagination
+      v-show="isRead === 2"
+      :total="feedBackStore.unReadFeedBacks.total"
+      :page="feedBackStore.unReadFeedBacks.current"
+      :limit="feedBackStore.readFeedBacks.size"
+      @pagination="clickToUnRead"
+      class="pagination_feedBack"
+    />
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { ref, onMounted, toRefs, computed } from 'vue'
+import { ElMessageBox, ElMessage, type TabsPaneContext } from 'element-plus'
+import { useFeedBackStore } from '@/store/modules/feedback'
+import { chooseContent } from '@/utils/chooseFeedBack'
+import { changeStateAPI } from '@/api/feedback'
+import { pageInfo } from '@/types/pageMessage'
+import { ReplyInfo } from '@/types/feedback'
+
+//获取数据
+const feedBackStore = useFeedBackStore()
+const activeName = ref('first')
+//打开反馈
+const dialogTableVisible = ref(false)
+
+//照片地址
+const url = ref()
+
+//预览照片地址
+const srcList = ref<string[]>([])
+
+//收到的反馈内容
+const feedBackContent = ref('')
+
+//文本框内容
+const textarea = ref('')
+
+//控制btn 按钮
+const replyStatus = ref([false])
+
+// 存储 userId
+const userId = ref(0)
+
+//表示pagination
+const isRead = ref(0)
+
+// 回复消息
+const handleMsg = async () => {
+  if (textarea.value === '') {
+    ElMessage.warning('回复内容不能为空')
+  } else {
+    // 调用后端接口
+    await feedBackStore.replyMessage(userId.value, textarea.value)
+
+    // 关闭回复框
+    dialogTableVisible.value = false
+
+    // 将 textarea 内容清空
+    textarea.value = ''
+  }
+}
+// 取消发送
+const cancelMsg = () => {
+  // 清空 textarea 内容
+  textarea.value = ''
+  // 关闭回复框
+  dialogTableVisible.value = false
+}
+
+//切换tab 栏
+const handleClick = (tab: TabsPaneContext) => {
+  // 提取tab 信息
+  let string = tab.props.label.substring(0, tab.props.label.indexOf('('))
+  if (string === '全部') {
+    // ChangeTabMsg(0)
+    isRead.value = 0
+    feedBackStore.unReadInfo(0, 10)
+  } else if (string === '已查看') {
+    isRead.value = 1
+    // 获取已读数据
+    feedBackStore.readInfo(0, 10)
+  } else {
+    isRead.value = 2
+    // 获取未读数据
+    feedBackStore.unReadInfo(0, 10)
+  }
+}
+
+//点击打开详情 (要抽离逻辑)
+const viewDetail = async (id: number, user: number) => {
+  // 控制详情页显示与否
+  dialogTableVisible.value = true
+  userId.value = user
+
+  //通过 id 筛选出对应反馈信息
+  let feedBacks = feedBackStore.userFeedBack.filter((item) => item.uuid === id)
+
+  // 视图更新
+  url.value = feedBacks[0]?.imageUrl[0]
```

Comment:
这里用 ?. 的地方有点多。你在查找后直接判断 feedbacks 是不是空，如果是空就 return 是不是就跟方便。用多个 ?. 首先很麻烦，其次影响性能
```suggestion
  if (!feedback) {
    return;
  }
  
  url.value = feedBack.imageUrl[0];
```